analyses/copy_number_consensus_call/src/scripts/restructure_column.py

```diff
@@ -0,0 +1,73 @@
+################## PURPOSE #################
+# This script is to restructure the 4th(start pos),5th(end pos), and 6th(copy number) columns of the input file
+# Because these columns hold the "raw" coordinates of the CNV after filtering out for IGLL, telomerics, centromeric
+# and segup regions, we want to retain these info for any analysis further down stream.
+# Thus this script format columns 4, 5, and 6 of a CNV into ---  start_pos:end_pos:copy_number  ----
+# So if a CNV is part of a FINAL consensus CNV, its (start_pos:end_pos:copy_number) information will
+# be carried through and contained in the FINAL FILE
+
+# For example
+#          column 4                   column 5             column 6
+# 2777350,6035598,6807639       6035597,6807638,7659545     3,3,3
+
+# Will become 2777350:6035597:3,6035598:6807638:3,6807639:7659545:3,
+################# ASSUMPTION ###############
+
+# The bed file has no header
+
+############################################
+
+# Imports in the pep8 order https://www.python.org/dev/peps/pep-0008/#imports
+# Standard library
+import argparse
+import sys
+import os
+
+# Related third party
+import numpy as np
+import pandas as pd
+
+## Define parser for the input file
+parser = argparse.ArgumentParser(description="""This script restructure the 4th, 5th, and 6th columns""")
+parser.add_argument('--file', required=True,
+                    help='path to the bed file that needs to be restructred')
+
+args = parser.parse_args()
+
+
+## Check to see if the bed file is empty, if so, make an empty dataframe 
+if os.stat(args.file).st_size == 0:
+    new_mat = pd.DataFrame(columns=['a','b'])
+
+## Use pandas to read in the bed file.
+## Since Manta doesn't have copy numbers and "NA" is used for Manta's copy numbers, we need
+## the "keep_default_na" so that the "NA"s don't get converted into NaN by pandas
+else:
+    file = pd.read_csv(args.file, delimiter='\t', header=None, keep_default_na=False)
+
+    ## Make a new column to store info in the new format (start_pos:end_pos:copy_number)
+    new_column = file.iloc[:,(0)].copy()
+
+    ## For every CNV in the file, reformatt columns 4, 5, and 6
+    for i_index,i in enumerate(file.itertuples()):
+        ## Since bed file uses a comma as default for collapsing information, we split the information using ','
+        list_start = str(i._4).split(',')
+        list_end = str(i._5).split(',')
+        list_cn = str(i._6).split(',')
+
+        ## Initialize a variable to store the new restructred information
+        growing_list = ''
+
+        ## Rearange the split information into the new format
+        for j in range(0,len(list_start)):
+            growing_list += '{}:{}:{},'.format(list_start[j],list_end[j],list_cn[j])
```

Comment:
@jashapiro I see! Thank you for the suggestion. I never knew about the zip() function. It is helpful to know it exists! Thanks for that. I will add that in no problem!! Just a quick question, did you add an extra `]` by accident at `format(*cnv])` ?

Reply:
yes, that is an accident!

Reply:
The * is not though, as you need to unpack the tuple.

Reply:
@jashapiro  Yep! I fixed it and since list comprehension output my info in a list, I added join() to combine all of my strings together. Please have a look.